soc/xtensa/intel_adsp/common/ipc.c

```diff
@@ -145,15 +156,15 @@ bool intel_adsp_ipc_send_message(const struct device *dev,
 	return true;
 }
 
-bool intel_adsp_ipc_send_message_sync(const struct device *dev,
+int intel_adsp_ipc_send_message_sync(const struct device *dev,
 				uint32_t data, uint32_t ext_data,
 				k_timeout_t timeout)
 {
 	struct intel_adsp_ipc_data *devdata = dev->data;
 
-	bool ret = intel_adsp_ipc_send_message(dev, data, ext_data);
+	int ret = intel_adsp_ipc_send_message(dev, data, ext_data);
 
-	if (ret) {
+	if (ret != 0 && ret != -ECANCELED) {
```

Comment:
you have changed the behavior here, if `intel_adsp_ipc_send_message()` returns `-EBUSY` you will execute this block ... 
The right is probably `if (ret == 0)`